### PR TITLE
Add react router 6 error elements

### PR DESCRIPTION
**Pull request by philipp-spiess**
I was looking into why the error boundaries are no longer working as expeced.

It turns out that RR6 no longer bubbles errors through to the React error boundary but instead using their own [`errorElement` API](https://reactrouter.com/en/main/route/error-element). 🙃  

To reproduce the previous behavior, this PR manually defines an `errorElement` for all Routes that are mounted in the same file that also mounts an `<ErrorBoundary />`. This way, the error does not unintentionally bubble to the root of the React router.

## Test plan

Tested manually, here's an error that bubbles to the top:

Screenshot 2023-02-27 at 16 41 39

And here an error inside the user area:

Screenshot 2023-02-27 at 16 51 22

## App preview:

- [Web](https://sg-web-ps-rr6-error-handling.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.01% (+0.41 kb) | 0.01% (+0.92 kb) | 0.00% (+0.51 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/ps/rr6-error-handling/compare-report.html;tab=live_object) for a full comparison between the commits 6773de9 and e50a05d or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/components/ErrorBoundary.test.tsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React from 'react'
 
 import { renderWithBrandedContext } from '@sourcegraph/wildcard/src/testing'
 
-import { ErrorBoundary } from './ErrorBoundary'
+import { ErrorBoundary, RouteError } from './ErrorBoundary'
```
Comment (fkling) [inline client/web/src/components/ErrorBoundary.test.tsx:6]: Doesn't seem to be used?

File: client/web/src/components/ErrorBoundary.tsx [hunk 2]
```diff
@@ -74,55 +75,91 @@ export class ErrorBoundary extends React.PureComponent<React.PropsWithChildren<P
...
             )
         }
 
         return this.props.children
     }
+}
 
-    private onReloadClick: React.MouseEventHandler<HTMLElement> = () => {
-        window.location.reload() // hard page reload
+export const RouteError: React.FC = () => {
```
Comment (fkling) [inline client/web/src/components/ErrorBoundary.tsx:92]: A doc comment that describes the purpose of this component would be useful.

File: client/web/src/team/TeamsArea.tsx [hunk 3]
```diff
@@ -42,10 +42,10 @@ const AuthenticatedTeamsArea: React.FunctionComponent<React.PropsWithChildren<Pr
         <ErrorBoundary location={location}>
```
Comment (fkling) [inline client/web/src/team/TeamsArea.tsx:42]: Does this still have any effect (i.e. would we need to keep it)? (same for other locations where we use it to wrap routes)

**Review by fkling — APPROVED** [review]
Approving to unblock.

File: client/web/src/team/TeamsArea.tsx (hunk 3 above)
Comment (fkling) [reply client/web/src/team/TeamsArea.tsx]: Definitely makes sense to have one or more boundaries that can catch errors outside or router. I'd only remove them where they are really not needed anymore.

**Review by valerybugakov — APPROVED** [review]
Nice, it was raised in Slack recently 👍

**Pull request merged**